compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -2334,6 +2334,25 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                 }
             }
 
+            Rvalue::ShallowInitBox(_, ty) => {
+                // Even with unsized locals cannot box an unsized value.
+                if self.unsized_feature_enabled() {
+                    let span = body.source_info(location).span;
+                    self.ensure_place_sized(ty, span);
+                }
+
+                let trait_ref = ty::TraitRef {
+                    def_id: tcx.require_lang_item(LangItem::Sized, Some(self.last_span)),
+                    substs: tcx.mk_substs_trait(ty, &[]),
+                };
+
+                self.prove_trait_ref(
```

Comment:
I guess this is copied from `NullaryOp` (can the branches be merged?), but I don't get why this is both doing `ensure_place_sized` *and* proving that `ty: Sized`

Reply:
Indeed. With a bit of archaeology I found that this check was introduced in #53575, and with #87935 the Sized constraint is verified in rustc_typeck so the error path will never be hit. The unsized path definitely could be removed.